lib/repos/remote/knowns.go

```diff
@@ -0,0 +1,116 @@
+package remote
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"sync"
+)
+
+var (
+	knownGit = []string{
+		"github.com",
+		"gitlab.com",
+		"bitbucket.com",
+	}
+	knownOCI = []string{}
+)
+
+const knownsFileName = "hof-knowns.json"
```

Comment:
the hof- here is redundant. I can see us expanding this to take patterns that map onto mirrors, including subpaths, so we can match regexp or something.

We'll still want some defaults here too ofc. But more to say, I can see what is stored in this file take on a different shape. So maybe a filename that might represent that better might be "mirrors.json". We should take an environment variable that explicitly the path to this file. I'd like to standardize around HOF_<AREA>_<THING>, for this HOF_MOD_MIRRORFILE.

We actually have a function that can load any datafile, so they can choose the format.

https://github.com/hofstadter-io/hof/blob/_dev/lib/datautils/io/io.go (There's more like it around in slightly different contexts, probably some inconsistency :/ )